### PR TITLE
Replaced pattern matching with if-else on boolean for compactness and performance

### PR DESCRIPTION
**Pull request by mrsushil118**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [ ] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [ ] Have you added copyright headers to new files?
* [ ] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [ ] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #xxxx

## Purpose

Improve scala code for compactness and performance.

### CODE REVIEW

File: framework/src/play/src/main/scala/play/api/http/HttpConfiguration.scala [hunk 1]
```diff
@@ -33,24 +33,11 @@ case class HttpConfiguration(
...
-  def serverDecoder: ServerCookieDecoder = strict match {
-    case true => ServerCookieDecoder.STRICT
-    case false => ServerCookieDecoder.LAX
-  }
-  def clientDecoder: ClientCookieDecoder = strict match {
-    case true => ClientCookieDecoder.STRICT
-    case false => ClientCookieDecoder.LAX
-  }
+case class CookiesConfiguration(strict: Boolean = true) {
+  def serverEncoder: ServerCookieEncoder = if (strict) ServerCookieEncoder.STRICT else ServerCookieEncoder.LAX
```
Comment (wsargent) [inline framework/src/play/src/main/scala/play/api/http/HttpConfiguration.scala]: given this is a configuration case class, overrides aren't at issue -- these can be made into `val` to save on execution.

Comment (mrsushil118) [inline framework/src/play/src/main/scala/play/api/http/HttpConfiguration.scala]: @gmethvin This class has not any public constructor, so we can not create a object with constructor.

```
 private ServerCookieEncoder(boolean strict) {
        super(strict);
    }
```

Comment (gmethvin) [inline framework/src/play/src/main/scala/play/api/http/HttpConfiguration.scala]: Okay, never mind then. It actually is our code so we could change it, but it's based on code from Netty and I don't see any point making this diverge from the original even more just to save a few characters.

**Comment by mrsushil118** [comment]
@gmethvin I closed this PR due to merged some unrelated changes. I created another new PR to reference this:
https://github.com/playframework/playframework/pull/6644

**Pull request closed without merging**